libs-scala/postgresql-testing/src/main/scala/com/digitalasset/testing/postgresql/PostgresAround.scala

```diff
@@ -10,54 +10,86 @@ import java.nio.file.{Files, Path}
 import java.util.UUID
 import java.util.concurrent.atomic.AtomicBoolean
 
+import com.daml.ports.Port
 import com.daml.testing.postgresql.PostgresAround._
 import org.apache.commons.io.{FileUtils, IOUtils}
 import org.slf4j.LoggerFactory
 
 import scala.collection.JavaConverters.asScalaBufferConverter
+import scala.util.Try
 import scala.util.control.NonFatal
 
 trait PostgresAround {
-  @volatile
-  private var fixture: PostgresFixture = _
+  @volatile private var server: PostgresServer = _
+  @volatile private var paths: Option[PostgresServerPaths] = None
 
   private val started: AtomicBoolean = new AtomicBoolean(false)
 
-  protected def startEphemeralPostgres(): Unit = {
+  protected def connectToPostgresqlServer(): Unit = {
+    (
+      sys.env.get("POSTGRESQL_HOST"),
+      sys.env.get("POSTGRESQL_PORT").map(port => Port(port.toInt)),
+      sys.env.get("POSTGRESQL_USERNAME"),
+      sys.env.get("POSTGRESQL_PASSWORD"),
+    ) match {
+      case (Some(hostName), Some(port), Some(userName), Some(password)) =>
```

Comment:
Making the decision here implicitly seems a bit scary. Can we at least add some logging statement what we are doing?

Reply:
It's logged [on line 48](https://github.com/digital-asset/daml/pull/5919/files#diff-0f0e5ff754ef51098020dfd7c5408183R48).

It'll look like this:

```
[INFO] Connected to PostgreSQL on localhost:54321.
```

Reply:
And [on line 53](https://github.com/digital-asset/daml/pull/5919/files#diff-0f0e5ff754ef51098020dfd7c5408183R53) for the other branch of the conditional.